torch/utils/_sympy/value_ranges.py

```diff
@@ -83,6 +83,19 @@ def __contains__(self, x):
         x = simple_sympify(x)
         return sympy_generic_le(self.lower, x) and sympy_generic_le(x, self.upper)
 
+    def tighten(self, other):
```

Comment:
What exactly does tighten do?

Reply:
i.e. given two bounds, it returns the AND of them.

Reply:
add docstring / type annotation ?